feat(search): Create my_teams filter with changes under assign-to-me feature flag

17hogeju opened the pull request:
Closes https://github.com/getsentry/sentry/issues/50842 and https://github.com/getsentry/sentry/issues/50847

Modifies `assigned_to_filter` query such that when the assign-to-me feature flag is enabled, the `me` filter only returns issues assigned to the user.

Also adds a filter `my_teams` that returns issues assigned to the user's teams and not the individual user.

Finally, adds tests for these changes

17hogeju committed:
feat(search): Modify assigned_to_filter query to separate user from teams

17hogeju committed:
feat(search): Add filter for my_teams

17hogeju committed:
feat(search): Add tests for my_teams filter

17hogeju commented on src/sentry/search/snuba/backend.py:
```diff
@@ -225,20 +226,10 @@ def assigned_or_suggested_filter(
     if "User" in types_to_owners:
         users = types_to_owners["User"]
         user_ids: List[int] = [u.id for u in users if u is not None]
-        team_ids = list(
```
Yes, you are right! Let me fix that

roggenkemper commented:
Not code related, but right now your PR has the title "Julia/assign to me" . i think it would be ideal to change it to be more in line with the commit messages you have (these don't matter as much , so no need to stick to the format here), just so it has the same format as other PRs in the repo.

barkbarkimashark commented on src/sentry/search/snuba/backend.py:
```diff
@@ -225,20 +226,10 @@ def assigned_or_suggested_filter(
     if "User" in types_to_owners:
         users = types_to_owners["User"]
         user_ids: List[int] = [u.id for u in users if u is not None]
-        team_ids = list(
-            Team.objects.filter(
-                id__in=OrganizationMemberTeam.objects.filter(
-                    organizationmember__in=OrganizationMember.objects.filter(
-                        user_id__in=user_ids, organization_id=organization_id
-                    ),
-                    is_active=True,
-                ).values("team")
-            ).values_list("id", flat=True)
-        )
```
I think we should keep this PR small and leave changes to `assigned_or_suggested` filter to another PR. What do you think?

17hogeju replied:
I brought that code back and put it under the feature flag. Removing the team_id from the query is necessary for the changes to work. Does that address your question?

barkbarkimashark replied:
Maybe I'm missing something, but I'm not sure changes to `assigned_or_suggested` are needed to get `assigned_to` filter to work? It might make sense to break up the PR so this PR is only changing code for `assigned_to/assigned` related functionality and open another PR for `assigned_or_suggested` changes. 

This has the benefit of keeping the scope of your PRs smaller and more manageable for reviewers to grok.

17hogeju replied:
I see what you mean. I have moved the logic to a new PR: https://github.com/getsentry/sentry/pull/51200

barkbarkimashark reviewed (commented):
Can we add a couple tests to https://github.com/getsentry/sentry/blob/d2e3acb22365fe65f327c799551749245fba281f/tests/snuba/search/test_backend.py#L1069-L1097

to make sure the before/after behavior of the feature flag works as we expect it?

barkbarkimashark commented:
Oh, you probably wanna change your PR description to follow the convention: https://develop.sentry.dev/commit-messages/ 

that doc is kinda a bit backwards for commit messages. Instead, I usually follow that convention for PR descriptions, and commit messages doesn't need to be as strict. I just write up a small sentence describing the commit.

17hogeju committed:
feat(search): Bring back deleted lines

17hogeju committed:
feat(search): Put query for team_ids behind feature flag

17hogeju committed:
feat(search): Nit: remove comments

17hogeju commented on src/sentry/search/snuba/backend.py:
```diff
@@ -79,24 +77,27 @@ def assigned_to_filter(
+                    f"{field_filter}__in": GroupAssignee.objects.filter(
+                        project_id__in=[p.id for p in projects],
+                        team_id__in=list(
+                            Team.objects.filter(
+                                id__in=OrganizationMemberTeam.objects.filter(
+                                    organizationmember__in=OrganizationMember.objects.filter(
+                                        user_id__in=user_ids,
+                                        organization_id=projects[0].organization_id,
+                                    ),
+                                    is_active=True,
+                                ).values_list("team_id", flat=True)
```
> Could we use the `get_teams_for_user` function here:
> 
> ```
> team_id__in=[team.id for team in get_teams_for_user()]
> ```

Will the `flat=True` still apply with the change you suggested above?

barkbarkimashark replied:
Yeah, we're basically flattening the returned data by iterating over the teams and grabbing the `id`.

17hogeju committed:
feat(search): Reduce redundancy

17hogeju committed:
feat(search): Remove uneeded new variable

17hogeju committed:
feat(search): Rename variable to match actual meaning

17hogeju committed:
feat(search): Revert changes to separate logic in PR

scttcper committed:
test(ui): Convert incident redirect to tsx (#51197)

17hogeju committed:
feat(search): Fix breaking query

evanpurkhiser committed:
ref(ts): Convert onboardingTasks.spec to tsx (#51097)

getsentry-bot committed:
ref: bump sentry-kafka-schemas to 0.1.12 (#51202)

Co-Authored-By: lynnagara <1779792+lynnagara@users.noreply.github.com>

Co-authored-by: getsentry-bot <10587625+getsentry-bot@users.noreply.github.com>
Co-authored-by: lynnagara <1779792+lynnagara@users.noreply.github.com>

17hogeju committed:
feat(search): Fix error by passing params and modify get_teams_for_users

evanpurkhiser committed:
ref(js): Consolidate dateTime format functions (#51201)

ryan953 committed:
chore: Extract requestIdleCallback shim to it's own file (#51212)

https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback

lynnagara committed:
meta: Update codeowners (#51206)

So Snuba team gets tagged on changes to the unified consumer

Commit:
feat(crons): Add tooltip to timeline component (#51179)

Adds a tooltip like this

<img width="650" alt="image"
src="https://github.com/getsentry/sentry/assets/9372512/a8b32ae5-68dc-4478-8f88-93abcac466e0">

Commit:
ref(dynamic-sampling): Cleanup tasks for dynamic sampling (#50721)

priscilawebdev committed:
feat(project-creation): Make product selection feature available to everyone - Part 3 (#50934)

ryan953 committed:
bug(replays): Fix typo in SlowClickFrameData type definition (#51211)

obostjancic committed:
test(onboarding): assert project platform (#51041)

Commit:
ref(dynamic-sampling): Remove unused project options (#51237)

Commit:
ref(dynamic-sampling): Turn off sliding window cron job (#51238)

priscilawebdev committed:
ref(hook): Hide product availability behind a feature flag (#51239)

getsentry-bot committed:
ref: bump sentry-arroyo to 2.12.0 (#51240)

Co-authored-by: untitaker <837573+untitaker@users.noreply.github.com>
Co-authored-by: getsentry-bot <10587625+getsentry-bot@users.noreply.github.com>

untitaker committed:
fix(buffers): Add dates serialization to json serializer (#51146)

Swatinem committed:
Split up backpressure.high_watermarks option (#51247)

This introduces a separate option for each service.

Fixes https://github.com/getsentry/team-processing/issues/62

Commit:
ref(dynamic-sampling): Improve compatibility check endpoint (#51242)

priscilawebdev committed:
ref(product-selection): Back with the question mark (#51249)

markstory committed:
feat(hybridcloud) Squash migrations (#50993)

With all the foreign keys between region and control silo models broken
we need to start testing the split database mode in CI and dev
environments. In order to split the database for these environments we
need:

1. Django database routers that understand silo assignments (done)
2. Migrations that don't have foreign key definitions that would be
invalid in the split database scenario. (these changes)

I've not removed any old migrations yet but will do that in a couple
weeks when most engineers have updated their environments.

getsentry sibling https://github.com/getsentry/getsentry/pull/10836

narsaynorath committed:
feat(alerts): Add new transaction metrics (#51162)

These are new metrics we're storing for transactions. We need to add
them to the supported list so we can create alerts on them.

Abdkhan14 committed:
feat(perf-detector-threshold-configuration) Re-added detection enabled/disabled code (#51248)

Last step in the data migration process.
New functionality was already approved here:
https://github.com/getsentry/sentry/pull/50825

- Manually set enabled/disabled values based on existing detection
rates.
- Removed detection_rate functionality.
- Used detect_enabled functionality.
- Added back, commented out tests. 
- Removed an overlapping test from test_performance_detection.py

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

DominikB2014 committed:
feat(starfish): issue list improvements (#51148)

This PR makes a few changes
1. Makes the issue list more closely match mocks by removing several
columns and adding a 'View All' button.
<img width="733" alt="image"
src="https://github.com/getsentry/sentry/assets/44422760/b8f49495-10a6-4af1-8043-7d5a9038520d">

2. Fixes a bug where issues always return no results because the date
condition was passed into the `query` queryParam instead of its own
query param.

3. Update the groupList component to accept custom columns,
@getsentry/issues if someone could have a look on your side that would
be great. This `withColumns` prop can deprecate the `withChart` prop as
it's more generic. I can go through the code and make the change from
`withChart` to `withColumns` if this PR is given ✅

---------

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

0Calories committed:
feat(starfish): Add per second unit to y axes of throughput charts (#51166)

wmak committed:
feat(starfish): Add span samples endpoint (#51170)

- This adds an endpoint that will for a given time range return sample
spans that are spread out both in duration and time
- Without any params the endpoint will determine the upper and lower
bounds to get duration itself from the metrics dataset
- Defaults to the span.self_time column, and defaults to 3 buckets for
time

Sample
call:
http://dev.getsentry.net:8000/api/0/organizations/sentry/spans-samples/?end=2023-04-28T00:00:00&start=2023-04-21T00:00:00&lowerBound=1&upperBound=2

Response: 
```json
{
  "data": [
    {
      "timestamp": "2023-04-21T00:38:31+00:00",
      "span.self_time": 1.904,
      "transaction.id": "0f46cd0af7824212bf92a1a1e9fc2158",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-22T11:23:35+00:00",
      "span.self_time": 8.476,
      "transaction.id": "1e36c596c2cb405c9030a70c9d8db7ba",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-22T11:25:26+00:00",
      "span.self_time": 0.011,
      "transaction.id": "1f873bee6db94ba3a2ea798793fc3172",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-24T23:32:37+00:00",
      "span.self_time": 0.019,
      "transaction.id": "2c2fa0c8f5ac48d6bd7fee9abc3ace76",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-25T00:43:09+00:00",
      "span.self_time": 1.074,
      "transaction.id": "3ab5501de48d487ba6d67081f7e578fe",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-25T00:57:04+00:00",
      "span.self_time": 2.324,
      "transaction.id": "1ae007a1f83544a2aa19092b84f335d4",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-26T11:09:53+00:00",
      "span.self_time": 3.062,
      "transaction.id": "08d146cffb8f435e981f5ac0691b8e5c",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-26T23:56:51+00:00",
      "span.self_time": 1.214,
      "transaction.id": "00f9a4020c0e4bd7bfb35071d5d01649",
      "project": "internal"
    },
    {
      "timestamp": "2023-04-27T00:07:11+00:00",
      "span.self_time": 0.009,
      "transaction.id": "05e6cf4e87ee40d08248a3f9a8f28ee5",
      "project": "internal"
    }
  ]
}

gggritso committed:
ref(starfish): Remove unused zerofilling code (#51257)

No more scraped data, no more scraped data utilities

jjbayer committed:
fix(txnames): Prevent scan of all redis keys (#50940)

Default environments do not define a dedicated redis cluster or redis
instance for the transaction name clusterer. Since we use `scan_iter` to
iterate through all keys in Redis, the clusterer becomes slower the
larger the cluster is overall, independent of how many keys we actually
hit.

To prevent scanning all keys, introduce a new "meta" set of project IDs.
Whenever a set of samples is newly created for a project, its ID is
added to this "meta" set in redis. When spawning clusterers, we iterate
only over this set instead.

This PR also unifies some of the helper functions of transaction name
and span description clustering, such that the change affects both.

mydea committed:
feat(org-tokens): Implement endpoints for OrgAuthToken (#51131)

This introduces the API endpoints for the new `OrgAuthToken` model.

Closes https://github.com/getsentry/sentry/issues/50932

mydea committed:
fix(org-tokens): Ensure all users can update token names (#51268)


ref https://github.com/getsentry/sentry/issues/50932

priscilawebdev committed:
ref(product-selection): Update props and styles (#51269)

loewenheim committed:
fix(backpressure): Catch and report some errors (#51270)

This catches two errors in backpressure monitoring:
1. If there's an exception while querying a service's memory usage, it
is reported and the service is considered unhealthy.
2. If there's an exception while recording service health in Redis, it
is reported.

untitaker committed:
fix: Fix some typing problems around dynamic sampling (#51256)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

armenzg committed:
feat(grouping): Upgrade projects to the latest grouping configuration (#51040)

This PR upgrades all projects that do not have auto-update disabled to
the 2023-01-11 grouping config. This will happen as events for the
project happen.

Customers that want to override [any of the values from the enhancers in
this
release](https://github.com/getsentry/sentry/blob/master/src/sentry/grouping/enhancer/enhancement-configs/newstyle%402023-01-11.txt)
can do so by using the project's [Stack Trace Rules under Issue
Grouping](https://docs.sentry.io/product/data-management-settings/event-grouping/stack-trace-rules/).
<img width="829" alt="image"
src="https://github.com/getsentry/sentry/assets/44410/fd06e3d3-8f7b-438d-9a96-8431aa511a0f">

For instance, this change:
`family:native package:/lib/** -app`
can be overwritten with:
`family:native package:/lib/** +app`

### Changes in this release
The
[2023-01-11](https://github.com/getsentry/sentry/blob/42a860c4fd13672cb3fa3384ede490654de44e0f/src/sentry/grouping/strategies/configurations.py#L189-L202)
group config has the following changes:
* A Java grouping fix:
[java_cglib_hibernate_logic](https://github.com/getsentry/sentry/pull/45185)
* [A bunch of stacktrace
enhancers](https://github.com/getsentry/sentry/blob/master/src/sentry/grouping/enhancer/enhancement-configs/newstyle%402023-01-11.txt)
(based on the mobile enhancers)
  * More info in section 2

#### Java change

From: https://github.com/getsentry/sentry/pull/45185 behind the
`java_cglib_hibernate_logic` branch:

Do not consider these as inApp:
- `$EnhancerBy...$`
- `$FastClassBy...$`
- `$HibernateProxy$`

Fixes https://github.com/getsentry/sentry-java/issues/1018

#### Stack Trace Rules Enhancers

[The
file](https://github.com/getsentry/sentry/blob/master/src/sentry/grouping/enhancer/enhancement-configs/newstyle%402023-01-11.txt)
has a bunch of enhancers. You can review the `pynsap` changes in the PR
to get a sense of the changes.

This is based off [the mobile enhancers
file](https://github.com/getsentry/sentry/blob/master/src/sentry/grouping/enhancer/enhancement-configs/mobile@2021-04-02.txt).
It basically adds the following [extra
rules](https://github.com/getsentry/sentry/blob/8ef3e68d626d06859960a630a46296d52a9b93dc/src/sentry/grouping/enhancer/enhancement-configs/newstyle%402023-01-11.txt#L404-L463).

All the PRs listed below are also part of [this
milestone](https://github.com/getsentry/sentry/milestone/50?closed=1).

The enhancements on top of the mobile enhancers includes these PRs (from
[this historical
view](https://github.com/getsentry/sentry/commits/master/src/sentry/grouping/enhancer/enhancement-configs/newstyle%402023-01-11.txt)):
* https://github.com/getsentry/sentry/pull/45423
* https://github.com/getsentry/sentry/pull/44761

The changes for the mobile enhancers can be seen
[here](https://github.com/getsentry/sentry/commits/master/src/sentry/grouping/enhancer/enhancement-configs/mobile%402021-04-02.txt):
* https://github.com/getsentry/sentry/pull/28669
* https://github.com/getsentry/sentry/pull/28069
* https://github.com/getsentry/sentry/pull/27994
* https://github.com/getsentry/sentry/pull/27880
* https://github.com/getsentry/sentry/pull/27836
* https://github.com/getsentry/sentry/pull/27744
* https://github.com/getsentry/sentry/pull/27781
* https://github.com/getsentry/sentry/pull/27754
* https://github.com/getsentry/sentry/pull/27314
* https://github.com/getsentry/sentry/pull/27239
* https://github.com/getsentry/sentry/pull/27076
* https://github.com/getsentry/sentry/pull/27197

---------

Co-authored-by: Armen Zambrano G <armenzg@users.noreply.github.com>
Co-authored-by: Arpad Borsos <arpad.borsos@sentry.io>

loewenheim committed:
fix(backpressure): Add timeout to RabbitMQ memory checks (#51274)

untitaker committed:
fix: Use multiprocessing strategy in ingest consumer tests (#51250)

17hogeju committed:
feat(search): Fix user type sent to get_teams_for_users

17hogeju commented:
Whoops, sorry for tagging everyone
